src/renderer/components/ImageCropper/index.tsx

```diff
@@ -103,11 +114,71 @@ export default function ImageCropper({
     })()
   }
 
+  const onZoomIn = () => {
+    zoom.current += 0.01
+    moveImages(posX.current, posY.current)
+  }
+
+  const onZoomOut = () => {
+    zoom.current -= 0.01
+    moveImages(posX.current, posY.current)
+  }
+
   const onZoomReset = () => {
+    if (!cutImage.current || !fullImage.current) {
+      return
+    }
     posX.current = 0
     posY.current = 0
     zoom.current = initialZoom.current
     moveImages(0, 0)
+    if (rotation.current > 0) {
+      cutImage.current.src = filepath
+      fullImage.current.src = filepath
+    }
+  }
+
+  const onRotateImages = () => {
+    if (!tmpCanvas.current || !fullImage.current || !cutImage.current) {
+      return
+    }
+
+    const canvas = tmpCanvas.current
+
+    const ctx = canvas.getContext('2d')
+
+    if (!ctx) {
+      console.log("Somehow there's no context for the image")
+      return
+    }
+
+    canvas.width = fullImage.current.clientHeight
+    canvas.height = fullImage.current.clientWidth
+
+    rotation.current += Math.PI * 0.5
+
+    // FIXME: we now only assume Math.PI * 0.5 rotations
+    if (rotation.current > Math.PI * 1.5) {
+      rotation.current = 0
```

Comment:
Although it appears not to, this looks like this condition could be `true` for when the rotation is actually 1.5 * π, although maybe I don't know enough about floating point numbers. Either way, the suggestion is better IMO.

```suggestion
    // FIXME: we now only assume Math.PI * 0.5 rotations
    if (rotation.current >= Math.PI * 1.999) {
      rotation.current = 0
```

Reply:
I'm not an expert on floating too, but I believe this is a pretty robust solution:
- JS actually uses double precision
- since we reset it to 0 on every full rotation nothing is accumulated, so if it's true once it will be true every time
- I don't think testing for equality gives us anything

I think if we someday go for the crazy idea of allowing free rotation, we might need a more informed solution, for now I believe current one is robust for the case we're going for.

Reply:
> if it's true once it will be true every time

Good point. Though I think my suggestion is more robust and less eyebrow-raising, so I'd still apply it, but I won't insist.